DDL.sql

```diff
@@ -0,0 +1,263 @@
+);
+
+CREATE TABLE TMSConfig(
+    ID SERIAL PRIMARY KEY,
+    name TEXT NOT NULL,
+    sensorTempUnits TEXT NOT NULL,
+    pumpSpeedUnits TEXT NOT NULL,
+    fanSpeedUnits TEXT NOT NULL
+);
+
+CREATE TABLE PVCConfig(
+    ID SERIAL PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE TABLE BikeBoardConfig(
+    bikeConfigID INTEGER,
+    boardConfig INTEGER,
+    PRIMARY KEY ( bikeConfigID, boardConfig)
+);
+
+CREATE TABLE BikePartConfig(
+    ID SERIAL PRIMARY KEY,
+    bikeConfigId INTEGER NOT NULL,
+    partNumber TEXT NOT NULL
+);
+
+CREATE TABLE Context(
+    ID SERIAL PRIMARY KEY,
+    date DATE NOT NULL,
+    runEventType TEXT NOT NULL,
+    airTemp INTEGER DEFAULT NULL,
+    altitude INTEGER DEFAULT NULL,
+    location TEXT NOT NULL,
+    windVelocity FLOAT DEFAULT NULL,
+    riderOne TEXT NOT NULL,
+    riderTwo TEXT DEFAULT NULL,
+    riderOneCurrentWeight FLOAT NOT NULL,
+    riderTwoCurrentWeight FLOAT DEFAULT NULL,
+    humidity FLOAT DEFAULT NULL,
+    driverFeedback TEXT DEFAULT NULL
+);
+
+CREATE TABLE BMSNumAttemptsMade(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSNumThermAttemptsMade(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP NOT NULL
+);
+
+CREATE TABLE BMSLastAttemptTime(
+    ID SERIAL PRIMARY KEY,
+    time INTEGER NOT NULL,
+    timeStamp TIMESTAMP NOT NULL,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSThermAttemptTime(
+    ID SERIAL PRIMARY KEY,
+    time INTEGER NOT NULL,
+    timeStamp TIMESTAMP NOT NULL,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSTotalVoltage(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSBatteryVoltage(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSCurrent(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSThermistorTemp(
+
+    ID SERIAL,
+    thermNumber INTEGER,
+    PRIMARY KEY (ID, thermNumber),
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSMinPackTemp(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER,
+    packId INTEGER,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSMaxPackTemp(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER,
+    packId INTEGER,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSBqInternalTemp(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSBqTemp(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER NOT NULL,
+    bqId INTEGER NOT NULL,
```

Comment:
I would call this tempID, not bqID. Both temps come from the same BQ chip. Also, you could probably combine this with table above if we just assume the internal temp is one of the BQ temps with a particular ID